### PR TITLE
[10.0] [REM] [IMP] Deprecate medical_base_history and update medical_appointment accordingly

### PR DESCRIPTION
**Pull request by obulkin**
[REM] medical_base_history
* Remove module, which was only used in ``medical_appointment`` and has been
deprecated in favor of ``auditlog``

[IMP] medical_appointment
* Fix broken ``medical.appointment`` unit test
* Remove all uses of deprecated module ``medical_base_history`` from manifest as
well as ``medical.appointment`` model and associated data files/views
* Add ``auditlog`` dependency and data file with ``auditlog.rule`` record for tracking
``medical.appointment`` changes
* Update ``medical.appointment`` to inherit from ``mail.thread`` for added tracking
and messaging functionality
* Update ``medical_appointment_view_form`` to show new ``mail.thread`` fields
* Remove comments field from ``medical.appointment`` model and associated data
files/views

### CODE REVIEW

**Comment by lasley** [comment]
@obulkin - tests are failing

**Comment by obulkin** [comment]
@lasley --- It's due to the ``auditlog`` dependency not being met: ``module medical_appointment: Unmet dependencies: auditlog``, possibly because it's in a different repo. How do I address that?

**Comment by lasley** [comment]
It's not merged yet right? You'll need to update the [oca_dependencies.txt](https://github.com/LasLabs/vertical-medical/blob/release/10.0/oca_dependencies.txt#L2)

**Comment by jcdrubay** [comment]
Auditlog is not available in v10 in oca: https://github.com/OCA/server-tools

But I just noticed that you did the migration on your server tools: https://github.com/LasLabs/server-tools/tree/10.0-mig-auditlog

oca_dependencies.txt should probably look like this:

```
partner-contact
# Waiting on merge of OCA/website#270
website https://github.com/laslabs/website.git release/10.0/website_field_autocomplete_related
server-tools https://github.com/laslabs/server-tools.git 10.0-mig-auditlog
```

Sorry for not following on that myself earlier  :/ and thanks for continuing :)

**Comment by lasley** [comment]
@obulkin - Auditlog was merged today, so no need to change oca_dependencies. I have restarted the builds.

File: medical_appointment/data/medical_appointment_data.xml [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
 
 <!--
-    Copyright (C) 2015-TODAY LasLabs [https://laslabs.com]
-    @author Dave Lasley <dave@laslabs.com>
-    @license AGPL-3
+    Copyright 2015-2017 LasLabs Inc.
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 -->
 
 <openerp>
```
Comment (lasley) [inline medical_appointment/data/medical_appointment_data.xml]: `<odoo>` tag. Also the two data tags can be removed to combine the sections - they're both noupdate=0

**Review by lasley — CHANGES_REQUESTED** [review]
@obulkin - Comment inline + please fix the warnings.

**Comment by obulkin** [comment]
@lasley - I've pushed up a commit that addresses your comment and the Runbot warnings.

File: medical_appointment/tests/medical_appointment.yml [hunk 2]
```diff
@@ -41,11 +41,11 @@
     email: arun@example.com
```
Comment (tedsalmon) [inline medical_appointment/tests/medical_appointment.yml]: File contains DOS line endings. Could you fix? This should do it for you in OS X:
```
perl -pe 's/\r\n|\n|\r/\n/g' medical_appointment.yml > medical_appointment.yml.fix && mv medical_appointment.yml.fix medical_appointment.yml
```

File: medical_appointment/views/medical_menu.xml [hunk 3]
```diff
@@ -6,19 +6,15 @@
     @license AGPL-3
 -->
 
-<openerp>
-    <data>
+<odoo>
```
Comment (tedsalmon) [inline medical_appointment/views/medical_menu.xml:9]: Space between Odoo and `menuitem`'s, please

**Review by tedsalmon — APPROVED** [review]
Approved pending minor changes requested. LGTM otherwise!

**Comment by lasley** [comment]
@tedsalmon - Oleg has been taking some classes, so may not be able to make the change in reasonable timeframe. Please just do yourself

**Comment by obulkin** [comment]
Thanks for pushing up the fix, @tedsalmon. I think this is ready for merge now, @lasley.

**Review by lasley — APPROVED** [review]

**Pull request merged**